### PR TITLE
refactor: controller-runtime v0.16.3

### PR DESCRIPTION
**Pull request by KevFan**
Resolves: https://github.com/Kuadrant/kuadrant-operator/issues/237

### CODE REVIEW

**Comment by KevFan** [comment]
Blocked / Kept as draft as decided to wait for MVP release before making these changes

File: main.go [hunk 1]
```diff
@@ -100,23 +102,25 @@ func main() {
...
+	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
+	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
+		"Enable leader election for controller manager. "+
+			"Enabling this will ensure there is only one active controller manager.")
 	flag.Parse()
 
-	options := ctrl.Options{Scheme: scheme}
-
-	if configFile != "" {
-		options, err = options.AndFrom(ctrl.ConfigFile().AtPath(configFile))
```
Comment (KevFan) [inline main.go:115]: `ComponentConfig` has been deprecated (https://github.com/kubernetes-sigs/controller-runtime/releases/tag/v0.15.0). We can either use a custom implementation that sets Manager.Options directly or move back to configuration via arguments. 

Moving back to configuration using arguments as it's the easier option and we do not have many configurations in general

**Comment by KevFan** [comment]
Gateway api v1.0.0 has been released using controller runtime v0.16.3
- https://github.com/kubernetes-sigs/gateway-api/blob/v1.0.0/go.mod#L17

**Review by adam-cattermole — APPROVED** [review]
These changes look good - I opted for a near identical implementation when I started to look at gwapi v1 and needed a newer controller-runtime. Nice work!

**Comment by KevFan** [comment]
Rebased to resolve conflicts with main

File: main.go [hunk 2]
```diff
@@ -100,23 +103,25 @@ func main() {
...
+		metricsAddr          string
+		enableLeaderElection bool
+		probeAddr            string
+		err                  error
 	)
-	flag.StringVar(&configFile, "config", "",
-		"The operator will load its initial configuration from this file. "+
-			"Omit this flag to use the default configuration values. "+
-			"Command-line flags override configuration from this file.")
+	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
```
Comment (eguzki) [inline main.go:111]: I wonder if the deployment of the operator should change after this change. Or defaults are good enough?

Comment (KevFan) [reply main.go:111]: I think these defaults are good. These are the same as https://github.com/Kuadrant/kuadrant-operator/commit/52b2cdb62e3594f39b353665428cf5531cce574c which was before it was moved to loading config using file.

Only difference I see between defaults and deployment options is that `leacterElection` is `true` only for deployment but `false` by default 🤔 :
- https://github.com/Kuadrant/kuadrant-operator/pull/246/files#diff-ef2df99e0c13ecb3c10ae6aa1923d7892fdccfa82e8f8777faebdaa4528c7568L1-L11
- https://github.com/Kuadrant/kuadrant-operator/pull/246/files#diff-319f8dc9adf6d477651de0cb44f85c968909535fc7c3a61be091a08b68298732R12

Comment (eguzki) [reply main.go:111]: But the deployment is setting it explicitly, so we should be good.

**Review by eguzki — APPROVED** [review]

**Pull request merged**